### ISSUE TITLE
dont work with multiple plugins

### ISSUE THREAD

**Issue by darlanmendonca** [meanie/mongoose-to-json#1]
I use another plugin, but when I add your plugin, they dont work.

my code:

```js
import mongoose from 'mongoose'
import privates from 'mongoose-private'
import mongooseToJson from 'meanie-mongoose-to-json'
import encode from '../encode/encode.helper.js'

const schema = new mongoose.Schema({
  email: {type: String, trim: true, required: true, unique: true},
  password: {type: String, required: true, set: encode.md5, private: true},
})

schema
  .plugin(privates)
  .plugin(mongooseToJson)

module.exports = mongoose.model('users', schema)
``

**Comment by adamreisnz** [comment]
@darlanmendonca thanks for reporting, I'll have a look at your example shortly.
I use multiple plugins myself as well, and they tend to work, but I assign them globally, e.g. `mongoose.plugin(require('meanie-mongoose-to-json'));`

**Comment by darlanmendonca** [comment]
I thanks too, for your plugin :D

hum, maybe be a conflict with my other plugin, mongoose-private.

when I remove the mongoose-private, your plugin works fine, as global, or in scheme.

but when I have mongoose-private, your plugin not work.

I test as global, in scheme, and reordering too :(

**Comment by adamreisnz** [comment]
@darlanmendonca I've had a look, and I think the problem actually lies with the `mongoose-private` plugin. There, the `toJSON` method gets overwritten which breaks the functionality of this package.

This package actually remembers if any other toJSON transformations have been defined on the schema, and calls them afterwards, to ensure that you can still do custom toJSON transformation.

*However*, I've gone ahead and implemented the `private` functionality in this toJSON plugin instead (v1.1.0).

So now you should only need this plugin:

```js
const mongoose = require('mongoose');
const toJson = require('meanie-mongoose-to-json');

const schema = new mongoose.Schema({
  email: {type: String, trim: true, required: true, unique: true},
  password: {type: String, required: true, private: true},
})

schema.plugin(toJson);

const User = mongoose.model('users', schema);
const user = new User({email: 'test@test.com', password: 'test'});

console.log(user.toJSON());
```

Outputs:

```
{ email: 'test@test.com', id: '5850459e2b49a0561efc72cd' }
```

Hope this helps!

**Comment by adamreisnz** [comment]
@darlanmendonca closing this issue now, let me know if that worked for you

**Comment by darlanmendonca** [comment]
thanks @adamreisnz, I'm working in that, after done, I say to you, where exactly conflict this :D

thank u very much!!

**Comment by abhisekp** [comment]
I've made a PR to fix this i.e. working this along with other plugins https://github.com/meanie/mongoose-to-json/pull/10

**Comment by adamreisnz** [comment]
I use this with many different plugins with no issues. Can you clarify the problem?

**Comment by abhisekp** [comment]
When I use this plugin before `mongoose-hidden` plugin, it throws error. If I add the `return ret` which is also non-breaking to add, everything works seamlessly.

**Comment by abhisekp** [comment]
This is in compliance with the transform function examples [Mongoose Transform](https://mongoosejs.com/docs/api/document.html#:~:text=We%20may%20need%20to%20perform%20a%20transformation%20of%20the%20resulting%20object%20based%20on%20some%20criteria,%20say%20to%20remove%20some%20sensitive%20information%20or%20return%20a%20custom%20object.)

**Comment by adamreisnz** [comment]
Thank you, this has been released as 2.5.0 👍
